mono/metadata/appdomain.c

```diff
@@ -379,27 +384,32 @@ mono_get_corlib_version (void)
 const char*
 mono_check_corlib_version (void)
 {
-	const char* res;
+	char *result = NULL;
```

Comment:
There is a reason for these 2 functions to be separated. Revert it.

Reply:
It just seems like extra layers to read through and nobody else uses the inner layer. It is unnecessary exposure, even if static. But ok.

Reply:
What is the reason? It should be in comments in the code. As the code stands, it seems like extra unnecesary functionality to provide both functions.